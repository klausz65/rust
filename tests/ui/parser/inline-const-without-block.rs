//@ run-rustfix

// See issue #78168.

#![feature(inline_const_pat)]

const fn one() -> i32 {
    1
}

fn foo() -> i32 {
    let x = 2;

    match x {
        const 2 => {}
        //~^ ERROR expected `{`, found `2`
        //~| HELP you might have meant to write this as part of a block
        _ => {}
    }

    match x {
        const 1 + 2 * 3 / 4 => {}
        //~^ ERROR expected `{`, found `1`
        //~| HELP you might have meant to write this as part of a block
        _ => {}
    }

    match x {
        const one() => {}
        //~^ ERROR expected `{`, found `one`
        //~| HELP you might have meant to write this as part of a block
        _ => {}
    }

    x
}

fn bar() -> i32 {
    let x = const 2;
    //~^ ERROR expected `{`, found `2`
    //~| HELP you might have meant to write this as part of a block

    x
}

fn baz() -> i32 {
    let y = const 1 + 2 * 3 / 4;
    //~^ ERROR expected `{`, found `1`
    //~| HELP you might have meant to write this as part of a block

    y
}

fn main() {
    foo();
    bar();
    baz();
}
