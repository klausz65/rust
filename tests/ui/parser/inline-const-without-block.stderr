error: expected `{`, found `2`
  --> $DIR/inline-const-without-block.rs:15:15
   |
LL |         const 2 => {}
   |               ^ expected `{`
   |
help: you might have meant to write this as part of a block
   |
LL |         const { 2 } => {}
   |               +   +

error: expected `{`, found `1`
  --> $DIR/inline-const-without-block.rs:22:15
   |
LL |         const 1 + 2 * 3 / 4 => {}
   |               ^ expected `{`
   |
help: you might have meant to write this as part of a block
   |
LL |         const { 1 + 2 * 3 / 4 } => {}
   |               +               +

error: expected `{`, found `one`
  --> $DIR/inline-const-without-block.rs:29:15
   |
LL |         const one() => {}
   |               ^^^ expected `{`
   |
help: you might have meant to write this as part of a block
   |
LL |         const { one() } => {}
   |               +       +

error: expected `{`, found `2`
  --> $DIR/inline-const-without-block.rs:39:19
   |
LL |     let x = const 2;
   |                   ^ expected `{`
   |
help: you might have meant to write this as part of a block
   |
LL |     let x = const { 2 };
   |                   +   +

error: expected `{`, found `1`
  --> $DIR/inline-const-without-block.rs:47:19
   |
LL |     let y = const 1 + 2 * 3 / 4;
   |                   ^ expected `{`
   |
help: you might have meant to write this as part of a block
   |
LL |     let y = const { 1 + 2 * 3 / 4 };
   |                   +               +

error: aborting due to 5 previous errors

